node-config-docsgen/action.yml

```diff
@@ -0,0 +1,159 @@
+        toolchain: ${{ inputs.rust-nightly-version }}
+        components: rustdoc
+
+    ## Generate configuration documentation
+    - name: Generate Configuration Documentation
+      id: generate_docs
+      shell: bash
+      working-directory: ./stacks-core
+      env:
+        OUTPUT_FILE: ${{ inputs.output }}
+        PROJECT_ROOT: $(pwd)
+      run: |
+        echo "Generating configuration documentation..."
+
+        # Navigate to the script directory and run
+        chmod +x contrib/tools/config-docs-generator/generate-config-docs.sh
+        bash contrib/tools/config-docs-generator/generate-config-docs.sh
+
+        # Verify output was generated
+        if [[ ! -f "$OUTPUT_FILE" ]]; then
+          echo "Error: Configuration documentation was not generated"
+          echo "Expected file: $OUTPUT_FILE"
+          exit 1
+        fi
+
+        echo "Documentation generated successfully at $OUTPUT_FILE"
+
+    ## Validate generated documentation quality
+    - name: Validate Generated Documentation
+      id: validate_docs
+      shell: bash
+      working-directory: ./stacks-core
+      run: |
+        echo "Validating generated documentation..."
+
+        OUTPUT_FILE="${{ inputs.output }}"
+
+        # Check if file exists and has content
+        if [[ ! -f "$OUTPUT_FILE" ]]; then
+          echo "Error: Markdown file not found at $OUTPUT_FILE"
+          exit 1
+        fi
+
+        # Check file size (should be substantial)
+        FILE_SIZE=$(wc -c < "$OUTPUT_FILE")
+        MIN_SIZE=${{ inputs.min-doc-size }}
+
+        if [[ $FILE_SIZE -lt $MIN_SIZE ]]; then
+          echo "Error: Generated documentation is too small ($FILE_SIZE bytes, minimum $MIN_SIZE)"
+          echo "This likely indicates a generation failure"
+          exit 1
+        fi
+
+        # Check word count and basic structure
+        WORD_COUNT=$(wc -w < "$OUTPUT_FILE")
+        LINE_COUNT=$(wc -l < "$OUTPUT_FILE")
+
+        echo "Documentation validation results:"
+        echo "  - File size: $FILE_SIZE bytes"
+        echo "  - Word count: $WORD_COUNT words"
+        echo "  - Line count: $LINE_COUNT lines"
+
+        # Basic content validation
+        if ! grep -q "# Configuration Reference" "$OUTPUT_FILE"; then
+          echo "Warning: Documentation may be malformed - missing main title"
+        fi
+
+        if [[ $WORD_COUNT -lt 1000 ]]; then
+          echo "Warning: Documentation seems very short ($WORD_COUNT words)"
+        fi
+
+        echo "Documentation validation completed successfully ✅"
+
+    ## Upload documentation as artifact
+    - name: Upload Documentation Artifact
+      id: upload_artifact
+      uses: actions/upload-artifact@ea165f8d65b6e75b540449e92b4886f43607fa02 # v4.6.2
+      working-directory: ./stacks-core
+      with:
+        name: node-parameters-bundle
+        path: ${{ inputs.output }}
+        retention-days: 30
+        if-no-files-found: error
+
+    ## Generate job summary
+    - name: Generate Job Summary
```

Comment:
It would be nice to capture some of the errors/warnings in the summary output vs reproducing it here. 
for example, this line:
```
          echo "Warning: Documentation may be malformed - missing main title"
```
would be great to have in the summary output. 


one idea is to pipe those warns/errors to a log that the summary step reads from and echoes vs. rewriting the previous step and removing the summary step.